lib/view_component/base.rb

```diff
@@ -97,25 +99,37 @@ def render_in(view_context, &block)
       @current_template = old_current_template
     end
 
+    # Called before rendering the component.
```

Comment:
```suggestion
    # Overridable lifecycle that is called before rendering the component.
```
Thoughts on mentioning that these are overridable lifecycle methods?